tests/bam_io/test_bam_index.h

```diff
@@ -75,39 +75,145 @@ SEQAN_DEFINE_TEST(test_bam_io_bam_index_open)
+    SEQAN_ASSERT(open(bamFile, toCString(bamFileName)));
+
+    BamIndex<Bai> baiFile;  // Open BAI index file
+    open(baiFile, toCString(baiFileName));
+
+    BamHeader header;
+    readHeader(header, bamFile);
+
+    std::vector<BamAlignmentRecord> records;
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 0);
+    SEQAN_ASSERT_EQ_MSG(length(records), 0u, "No reads expected in region seq1:1-0");
+    viewRecords(records, bamFile, baiFile, 0, 2000, 100000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 0u, "No reads expected in region seq1:2000-100000");
+    viewRecords(records, bamFile, baiFile, 2, 1, 2000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 0u, "No reads expected on invalid contigs.");
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 1);
+    SEQAN_ASSERT_EQ_MSG(length(records), 1u, "1 read expected in region seq1:1-1[");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 0);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 10);
+    SEQAN_ASSERT_EQ_MSG(length(records), 5u, "5 reads expected in region seq1:1-10");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 0);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 8);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 100);
+    SEQAN_ASSERT_EQ_MSG(length(records), 39u, "39 reads expected in region seq1:1-100");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 0);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 99);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 2000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 1501u, "1501 reads expected in region seq1:1-2000");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 0);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 1534);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 0, 100, 2000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 1472u, "1472 reads expected in region seq1:100-2000");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 66);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 1534);
+    clear(records);
+
+    // There is an unaligned read at position seq1:100 that should be missed in region [101,2000[
+    viewRecords(records, bamFile, baiFile, 0, 101, 2000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 1471u, "1471 reads expected in region seq1:101-2000");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 66);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 1534);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 0, 102, 2000);
+    SEQAN_ASSERT_EQ_MSG(length(records), 1470u, "1470 reads expected in region seq1:101-2000");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 69);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 1534);
+    clear(records);
+
+    viewRecords(records, bamFile, baiFile, 1, 300, 400);
+    SEQAN_ASSERT_EQ_MSG(length(records), 196u, "196 reads expected in region seq2:300-400");
+    SEQAN_ASSERT_EQ(front(records).beginPos, 265);
+    SEQAN_ASSERT_EQ(back(records).beginPos, 398);
+    clear(records);
+}
+
```

Comment:
newline detected by @rrahn, your private investigator you can trust :male_detective: